src/generators/AbstractGenerator.ts

```diff
@@ -274,6 +285,22 @@ export abstract class AbstractGenerator<
     return rawInputModel;
   }
 
+  protected async getRawInputModel(
+    input: any | InputMetaModel
+  ): Promise<InputMetaModel> {
+    const inputProcessorOptions = this.options.inputProcessor
+    switch (inputProcessorOptions?.type) {
+      case InputType.FILE:
+        const parser = new Parser();
+        const { document, diagnostics } = await fromFile(parser, input).parse();
+        // TODO : Want to log the diagnostics
+        if (!document) throw new Error('Invalid file input')
+        return this.process(document as any)
```

Comment:
This means that you can move this logic to the AsyncAPI processor where it belongs, because what if some of the other input processors want to support file input as well? Here we hardcode always to parse the input as AsyncAPI which is not the case 🙂